src/components/UserInfo.tsx

```diff
@@ -0,0 +1,23 @@
+import { User } from '../types/User';
+
+type Props = {
+  userData: User | null
```

Comment:
```suggestion
  userData: User
```
Якщо використати conditional rendering рівнем вище, то сюди завжди буде прилітати userData,  і вона ніколи не буде null